nanoAOD/plugins/HadTruthProducer.cc

```diff
@@ -35,23 +36,28 @@ HadTruthProducer::produce( edm::Event& iEvent, const edm::EventSetup& iSetup)
 
   edm::Handle<edm::View<reco::GenParticle>> genParticles;
   iEvent.getByToken(genLabel_, genParticles);
-
+  
+  edm::Handle<std::vector<std::vector<int>>> hadronIndices;
+  iEvent.getByToken(hadronIndices_, hadronIndices);
+  
   vector<int> nmatchedv;
   vector<int> isHadFromTsb;
   vector<uint8_t> isHadFromTop;
 
   std::map<const reco::GenParticle*, const reco::VertexCompositeCandidate*> matchHad; 
+  std::vector<const reco::GenParticle*> matchedGen;
+
   for (auto& cand : *hadronCands) {
+    if (abs(cand.pdgId()) == 5122) continue;
```

Comment:
This probably won't matter for you since Lambda_b is always after particles your interested in, but might if we adapt the code for D*, but shouldn't you push a nullptr onto matchedGen here to keep the matchedGen array consistent with the other arrays.